### PR TITLE
VY-52 Upload employee photo in s3 bucket

### PR DESCRIPTION
**Pull request by prkandel**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -58,6 +58,18 @@
 		</dependency>
 
 		<dependency>
+			<groupId>org.jboss.resteasy</groupId>
```
Comment (anishmanandhar) [inline pom.xml]: It is ok for now.. but try to make application as independent from app server as possible

File: src/main/java/com/lftechnology/vyaguta/core/config/S3Credential.java [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+import com.amazonaws.auth.profile.ProfileCredentialsProvider;
+import com.amazonaws.services.s3.AmazonS3;
+import com.amazonaws.services.s3.AmazonS3Client;
+
+public class S3Credential {
+
+    private static S3Credential instance;
+    private AWSCredentials credentials;
+    private AmazonS3 s3client;
+    private String configFile = "config.properties";
```
Comment (anishmanandhar) [inline src/main/java/com/lftechnology/vyaguta/core/config/S3Credential.java]: you can make this static constant

File: src/main/java/com/lftechnology/vyaguta/core/service/impl/EmployeeServiceImpl.java [hunk 3]
```diff
@@ -44,20 +48,42 @@ public Employee merge(UUID id, Employee obj) {
...
     }
 
     @Override
     public void remove(Employee entity) {
+        String bucketName = S3Credential.instance().getBucketName();
+        String imageFolderName = S3Credential.instance().getImageFolderName();
+        String keyName = entity.getFirstName()+"-"+entity.getLastName()+"-"+entity.getEmployeeNo()+".png";
+        AmazonS3 s3client = S3Credential.instance().getS3client();
+        
+        S3BucketUtil.deleteFile(bucketName, imageFolderName, keyName, s3client);
```
Comment (anishmanandhar) [inline src/main/java/com/lftechnology/vyaguta/core/service/impl/EmployeeServiceImpl.java]: move this to any archived folder

File: src/main/java/com/lftechnology/vyaguta/core/service/impl/FileUploadServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+                    InputStream inputStream = inputPart.getBody(InputStream.class, null);
+                    String keyName = employee.getFirstName() + "-" + employee.getLastName() + "-" + employee.getEmployeeNo() + ".png";
+                    imageUrl = S3BucketUtil.uploadImage(bucketName, imageFolderName, keyName, S3Credential.instance().getS3client(),
+                            inputStream);
+                } else {
+                    throw new ParameterFormatException("Invalid File format");
+                }
+            } catch (IOException e) {
+                // TODO change exception type
+                throw new DataAccessException("Unable to upload file");
```
Comment (anishmanandhar) [inline src/main/java/com/lftechnology/vyaguta/core/service/impl/FileUploadServiceImpl.java]: DataAccessException is not eligible here

File: src/main/java/com/lftechnology/vyaguta/core/util/S3BucketUtil.java [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+        for (S3ObjectSummary file : fileList) {
+            s3client.deleteObject(bucketName, file.getKey());
+        }
+        s3client.deleteObject(bucketName, folderName);
+    }
+    
+    public static void deleteFile(String bucketName, String folderName, String keyName, AmazonS3 s3client) throws AmazonServiceException{
+        String key = folderName + "/" + keyName;
+        s3client.deleteObject(bucketName, key);
+    }
```
Comment (anishmanandhar) [inline src/main/java/com/lftechnology/vyaguta/core/util/S3BucketUtil.java]: Favor Dto class over multiple parameters

**Comment by bhuwang** [comment]
Can one of the admins verify this patch?

**Comment by bhuwang** [comment]
Can one of the admins verify this patch?

**Pull request merged**